extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ConfigBuildStep.java

```diff
@@ -185,47 +195,43 @@ void validateConfigInjectionPoints(ValidationPhaseBuildItem validationPhase,
                     propertyDefaultValue = defaultValue.asString();
                 }
 
-                configProperties.produce(new ConfigPropertyBuildItem(propertyName, injectedType, propertyDefaultValue));
+                // @Observes @Initialized(ApplicationScoped.class) requires validation at static init
+                if (injectionPoint.getTarget().kind().equals(METHOD)
+                        && observerMethods.contains(injectionPoint.getTarget().asMethod())) {
```

Comment:
The `observerMethods.contains(injectionPoint.getTarget().asMethod())` matches any observer method, not only `@Observes @Initialized(ApplicationScoped.class)`. Is that expected?